src/lib/core/CHIPConfig.h

```diff
@@ -655,23 +655,31 @@
 #endif // CHIP_CONFIG_PEER_CONNECTION_POOL_SIZE
 
 /**
- * @def CHIP_PEER_CONNECTION_TIMEOUT_MS
+ * @def CHIP_CONFIG_SESSION_RECOVERY_BACKOFF_INITIAL_MS
  *
- * @brief After what period of inactivity is a peer connection considered
- * expired.
+ * @brief Initial backoff timeout for session recovery. Default 30 seconds
  */
-#ifndef CHIP_PEER_CONNECTION_TIMEOUT_MS
-#define CHIP_PEER_CONNECTION_TIMEOUT_MS 120000
-#endif // CHIP_PEER_CONNECTION_TIMEOUT_MS
+#ifndef CHIP_CONFIG_SESSION_RECOVERY_BACKOFF_INITIAL_MS
+#define CHIP_CONFIG_SESSION_RECOVERY_BACKOFF_INITIAL_MS (30 * 1000)
+#endif // CHIP_CONFIG_SESSION_RECOVERY_BACKOFF_INITIAL_MS
 
 /**
- * @def CHIP_PEER_CONNECTION_TIMEOUT_CHECK_FREQUENCY_MS
+ * @def CHIP_CONFIG_SESSION_RECOVERY_BACKOFF_MULTIPLIER
  *
- * @brief How frequent are peer connections checked for timeouts.
+ * @brief A multiplier which is used to increase exponential backoff timeout.
  */
-#ifndef CHIP_PEER_CONNECTION_TIMEOUT_CHECK_FREQUENCY_MS
-#define CHIP_PEER_CONNECTION_TIMEOUT_CHECK_FREQUENCY_MS 5000
-#endif // CHIP_PEER_CONNECTION_TIMEOUT_CHECK_FREQUENCY_MS
+#ifndef CHIP_CONFIG_SESSION_RECOVERY_BACKOFF_MULTIPLIER
+#define CHIP_CONFIG_SESSION_RECOVERY_BACKOFF_MULTIPLIER (1.3)
+#endif // CHIP_CONFIG_SESSION_RECOVERY_BACKOFF_MULTIPLIER
+
+/**
+ * @def CHIP_CONFIG_SESSION_RECOVERY_BACKOFF_MAX_MS
+ *
+ * @brief Max backoff timeout for session recovery. Default 5 minutes.
+ */
+#ifndef CHIP_CONFIG_SESSION_RECOVERY_BACKOFF_MAX_MS
+#define CHIP_CONFIG_SESSION_RECOVERY_BACKOFF_MAX_MS (5 * 60 * 1000)
+#endif // CHIP_CONFIG_SESSION_RECOVERY_BACKOFF_MAX_MS
```

Comment:
I am concerned about the potentially large stackup of retries at multiple layers here.  We already have MRP at the exchange layer for CASE.  And here I think we are multiplying that at another layer with session recovery, sometimes without resolution of a new address.

Won't this lead to extremely slow recovery in the case of renumbering?